test/block_tags.js

```diff
@@ -35,128 +34,77 @@ var contract = {
 };
 
 describe("Block Tags", function() {
-  var accounts;
-  var web3 = new Web3(Ganache.provider());
-  var contractAddress;
-
-  var initialBlockNumber;
-  var initial = {};
-
-  before("Gather accounts", function(done) {
-    web3.eth.getAccounts(function(err, accs) {
-      if (err) {
-        return done(err);
-      }
-      accounts = accs;
-      done();
-    });
+  const options = {
+    mnemonic: "candy maple velvet cake sugar cream honey rich smooth crumble sweet treat",
+    time: new Date(0) // Testing features that rely on determinate conditions
+  };
+  const web3 = new Web3(Ganache.provider(options));
+  const initial = {};
+  let accounts;
+  let contractAddress;
+  let initialBlockNumber;
+
+  before("Gather accounts", async function() {
+    accounts = await web3.eth.getAccounts();
   });
 
-  before("Get initial block number", function(done) {
-    web3.eth.getBlockNumber(function(err, n) {
-      if (err) {
-        return done(err);
-      }
-      initialBlockNumber = to.number(n);
-      done();
-    });
+  before("Get initial block number", async function() {
+    initialBlockNumber = to.number(await web3.eth.getBlockNumber());
   });
 
-  before("Get initial balance and nonce", function(done) {
-    async.parallel(
-      {
-        balance: web3.eth.getBalance.bind(web3.eth, accounts[0]),
-        nonce: web3.eth.getTransactionCount.bind(web3.eth, accounts[0])
-      },
-      function(err, result) {
-        if (err) {
-          return done(err);
-        }
-        initial = result;
-        initial.nonce = to.number(initial.nonce);
-        done();
-      }
-    );
+  before("Get initial balance and nonce", async function() {
+    const [balance, nonce] = [
+      ...(await Promise.all([
```

Comment:
I don't think we need to destructure the array here. Once awaited the result is already an array of resolved values (not pending Promises).

For example:

```
const [a, b] = await Promise.all([
  new Promise((resolve)=>{resolve("A")}),
  new Promise((resolve)=>{resolve("B")})
]);
console.log(a, b);
```
outputs `"A"`, `"B"`

Reply:
Agreed!  :100: